server/controllers/riderControllers.js

```diff
@@ -13,10 +13,16 @@ const findNearbyRoutes = async (req, res) => {
   const maxLat = JSON.parse(req.query.riderLocation).lat + .35;
   const minLng = JSON.parse(req.query.riderLocation).lng - .35;
   const maxLng = JSON.parse(req.query.riderLocation).lng + .35;
+  const userId = JSON.parse(req.query.userId);
+  console.log(`UserId: ${userId} is ${typeof userId}`);
+
   try {
     const nearbyRoutes = await Routes.findAll({
       where: {
-        [Op.and]: [{ latPickUp: { [Op.between]: [minLat, maxLat] } }, { lngPickUp: { [Op.between]: [minLng, maxLng] } }]
+        driver_id: { [Op.not]: userId},
+        rider_id: {[Op.or]: [userId, null]},
+        latPickUp: { [Op.between]: [minLat, maxLat] },
```

Comment:
ah this looks like a great way to filter routes...been doing it in a more convoluted way on the driver end but this is way cleaner!